libazureinit/src/kvp.rs

```diff
@@ -0,0 +1,649 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
We should move both this file and the `tracing.rs` file this PR adds to the `azure-init` package instead of `libazureinit` for the moment, for two reasons.

First, I'm not sure we want to expose the layer publicly in the `libazureinit` API. If it's a generally useful tracing layer for KVP it probably makes sense to be its own crate, but I think we should give it a little bit of time to bake before we do something like that. 

Secondly, libraries should not be configuring tracing subscribers. Libraries can use `tracing` and then it's up to the consuming application how to configure the way those logs are formatted/filtered/output.